internal/api/auth/callback.go

```diff
@@ -370,3 +383,25 @@ func adminGroup(groups []string) bool {
 	// ∴ user is not an admin.
 	return false
 }
+
+// allowedGroup returns true if one of the given OIDC
+// groups is equal to at least one allowed OIDC group.
+func allowedGroup(groups []string) bool {
+	if len(config.GetOIDCGroups()) == 0 {
+		// If no groups are configured, allow access (for backwards compatibility)
+		return true
+	}
+	for _, ag := range config.GetOIDCGroups() {
+		for _, g := range groups {
+			if strings.EqualFold(ag, g) {
+				// This is a group,
+				// ∴ user is allowed to log in
+				return true
+			}
+		}
+	}
+
+	// User is in no allowed groups,
+	// ∴ user is not allowed to log in
+	return false
```

Comment:
You can do this a bit nicer I think, using one of the new `slices` functions:

```suggestion
	allowedGroups := config.GetOIDCGroups()
	if len(allowedGroups) == 0 {
		// If no groups are configured as allowed,
		// allow access (for backwards compatibility).
		return true
	}

	// OK if one of claimed groups == an allowed group.
	for _, claimedGroup := range groups {
		if slices.ContainsFunc(allowedGroups, func(allowedGroup string) bool {
			return strings.EqualFold(claimedGroup, allowedGroup)
		})
	}

	// User is in no allowed groups,
	// ∴ user is not allowed to log in
	return false
```

Reply:
Should I also apply that change to the same logic in `adminGroup`?

Reply:
Sure!